frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -622,9 +622,15 @@ function getDatasetQueryParams(datasetQuery = {}) {
     type,
     query,
     native,
-    parameters: parameters.map(parameter => ({
-      ...parameter,
-      value: parameter.value ?? null,
-    })),
+    parameters: parameters
+      .map(parameter => ({
+        ...parameter,
+        value: parameter.value ?? null,
+      }))
+      .sort(sortById),
```

Comment:
When a user changes tabs, it checks to see if there is already a result for a card as well as if the last result contains all the data needed by the new settings of the card (parameters did not change from the last result). This check was failing in some cases just because the sort order of the parameters array was different. All the elements were present and equal, only the sort order was different.

Because the sort order has no effect on the desired behaviors, I added sorting to the parameters property to fix this issue.

- See [data-fetching.js#L333-L339](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/dashboard/actions/data-fetching.js#L333-L339) for where this check is occurring.